ConnectorRhino/ConnectorRhino/ConnectorRhinoShared/UI/MappingBindingsRhino.cs

```diff
@@ -26,41 +26,44 @@ public MappingBindingsRhino()
 
   public override MappingSelectionInfo GetSelectionInfo()
   {
-    try
+    List<RhinoObject> selection = RhinoDoc.ActiveDoc.Objects.GetSelectedObjects(false, false)?.ToList();
```

Comment:
legitimate I assume here to, that we expect each of these references to not be null?

Reply:
Can confirm that `.Objects` should never return null, but it's definitely not great to use `ActiveDoc` as per rhino documentation: `WARNING!! Do not use the ActiveDoc if you don't have to. Under Mac Rhino the ActiveDoc can change while a command is running. Use the doc that is passed to you in your RunCommand function or continue to use the same doc after the first call to ActiveDoc.` 

This it out of scope for this ticket however so I'm leaving it as is unless @AlanRynne knows if the call to `ActiveDoc` would ever return null (eg when called from rhino.inside?)

Reply:
It certainly feels like connector/converter logic requires a context object which would contain the document in context for the current calling context...  I need to play around with this in autofac so we can embody this in some DI in the future

Reply:
ActiveDoc **can and will be null** when running on `Rhino.Compute`, among other scenarios.

The need for a context object is real, there is contextual information that the connector needs to pass to the converter for it to do it's work properly (document, conversion caches, etc...) and some sub-conversions require different contextual information (i.e. Units.None instead of document units) to do their work properly.